src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
No, the callbacks are never called synchronously. They are always called after at least one animation frame passes.  If we use then, there's the digest needed.
It turns out it's simply a problem with the test setup. I'll post details later.